### PR TITLE
Sites landing page

### PR DESCRIPTION
**Pull request by jsernaque**

### CODE REVIEW

File: src/pages/guides/sites_frameworks.md [hunk 1]
```diff
@@ -2,25 +2,37 @@
...
+  If you are still planning or developing your headless Jamstack application, then consider using our [Traffic Splitting](/guides/performance/traffic_splitting) capability to migrate to headless Jamstack in stages.
 
-    <Callout type="info">
+</Callout>
 
-      {{ PRODUCT }} provides high availability for all of its components. You can choose a particular [region](/guides/performance#high-availability) of the world in which your API servers are located. {{ PRODUCT }} will provision two data centers closest to it, connecting them with automatic DNS failover. High availability is also provided within the data centers themselves, with all individual processes being (at least) duplicated and sharing the traffic load among themselves. This architecture minimizes the chances of traffic loss.
+## Regions {/*regions*/}
 
-    </Callout>
+{{ PRODUCT }} is available in the following global regions:
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:20]: I think this should be `{{ PRODUCT }} {{ PRODUCT_PLATFORM }}`

**Review by RaeesBhatti — COMMENTED** [review]
I think this is a good start but it might be useful to reimagine the homepage for Edgio sites docs. For example, do the sections like Caveats and Serverless Bundle Size belong on this page? Or should we spin those as separate pages?

**Comment by jsernaque** [comment]
> I think this is a good start but it might be useful to reimagine the homepage for Edgio sites docs. For example, do the sections like Caveats and Serverless Bundle Size belong on this page? Or should we spin those as separate pages?

I moved that information off to a separate article called "Limits."

File: src/pages/guides/sites_frameworks.md [hunk 2]
```diff
@@ -2,213 +2,39 @@
...
 ---
 
-{{ PRODUCT }} {{ PRODUCT_PLATFORM }} allows you to quickly integrate your headless Jamstack applications through one of our framework integrations. This integration allows {{ PRODUCT }} to further improve your website's performance through the server-side rendering (SSR) of your website's JavaScript. 
+{{ PRODUCT }} {{ PRODUCT_PLATFORM }}, which provides optimal performance for your headless Jamstack applications, is powered by [{{ PRODUCT }} {{ PRODUCT_EDGE }}](/guides/performance). This means that you will be able to develop your application at a faster rate and it empowers you to define performance optimizations within your application code. Build upon these performance improvements through static page generation:
+-   At build time through Static Site Generation (SSG).
+-   After your site has been built through [Incremental Static Regeneration (ISR)](/guides/sites_frameworks/isg).
+-   When a client requests a page through Server-Side Rendering (SSR). SSR generates static pages through our Serverless Compute workers. In addition to either eliminating the need for web servers or reducing their load, this improves performance by processing requests closer to your users.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:8]: We should change this to focus more on SSR rather than SSG. Since other platforms were built to cater to SSG use cases and are relatively poor at SSR use cases and our platform is stronger at SSR comparatively.
In general, SSR is preferred by users over SSG because it offers similar time-to-first-byte performance through caching but is inherently dynamic.

> SSR generates static pages through our Serverless Compute workers. In addition to either eliminating the need for web servers or reducing their load, this improves performance by processing requests closer to your users.

I think we need to rephrase the above, SSR doesn't really generate static pages, that's ISR. And replaces web servers, doesn't reduce their load. The SSR requests are not processed closer to the users necessarily. They can be cached closer to the users but that's the case for SSG and ISR as well.

File: src/pages/guides/sites_frameworks.md [hunk 3]
```diff
@@ -2,213 +2,39 @@
...
 
-## Availablity {/*availablity*/}
+## Getting Started
 
-{{ PRODUCT_NAME }} has multiple global regions in which it will automatically provision the following:
+{{ PRODUCT }} {{ PRODUCT_PLATFORM }} provides over [40 popular frameworks integrations](/guides/sites_frameworks/getting_started) through which you can deploy your headless Jamstack applications to {{ PRODUCT }}. Getting started with {{ PRODUCT }} {{ PRODUCT_PLATFORM }} involves initializing your {{ PRODUCT }} property through our CLI. Our CLI will then automatically detect your framework and configure your app accordingly.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:12]: > you can deploy your headless Jamstack applications

Let's rephrase this to "websites and apps" or "webapps" or something similar. We don't want to give the impression that our platform enables you to develop mobile/native applications.

Comment (RaeesBhatti) [reply src/pages/guides/sites_frameworks.md:12]: > property

Are we sure we want to call this property? If so, then it's okay. But let's check with the console team on what they'll be calling it. I know that it's currently called "site".

File: src/pages/guides/sites_frameworks.md [hunk 4]
```diff
@@ -2,213 +2,39 @@
...
+  If you are still planning or developing your headless Jamstack application, then consider using our [Traffic Splitting](/guides/performance/traffic_splitting) capability to migrate to headless Jamstack in stages.
 
-      {{ PRODUCT }} provides high availability for all of its components. You can choose a particular [region](/guides/performance#high-availability) of the world in which your API servers are located. {{ PRODUCT }} will provision two data centers closest to it, connecting them with automatic DNS failover. High availability is also provided within the data centers themselves, with all individual processes being (at least) duplicated and sharing the traffic load among themselves. This architecture minimizes the chances of traffic loss.
+</Callout>
 
-    </Callout>
+## Regions {/*regions*/}
 
--   A level 2 cache to maximize the global cache hit rate and reduce traffic to your servers.
+{{ PRODUCT }} {{ PRODUCT_PLATFORM }} is available in the following global regions:
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:22]: Let's make sure that the readers understand the these are the regions where the compute is available; the cache will be distributed to 100s of PoPs around the world, closer to end users.

File: src/pages/guides/sites_frameworks.md [hunk 5]
```diff
@@ -2,213 +2,39 @@
...
-
-```js
-// {{ DOCS_URL }}/guides/basics/edgio_config
-module.exports = {
-  includeFiles: require('./getNodeModules'),
-}
-```
+-   Running your code within two separate data centers.
+-   Setting up automatic DNS failover between those data centers.
+-   Load balancing computing requests between redundant processes.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:39]: Not sure about this one. I don't think we're doing any load balancing based on usage. @phdunham can you confirm please!

File: src/pages/guides/sites_frameworks/limits.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+-   You may need to [manually include NodeJS addons (aka native extensions).](#nodejs-native-extensions)
+-   Your code will only be granted [read-only access to the file system.](#readonly-filesystem-in-serverless-runtime)
+-   [Your project's bundle size](#serverless-bundle-size-limitation) cannot exceed 50 MB compressed or 250 MB uncompressed.
+-   Our serverless functions have a maximum runtime of 20 seconds per request. The response for a function that exceeds this limit is a [539 Project Timeout](/guides/performance/response#exclusive-status-codes).
+-   Our Serverless Compute workers are allowed to generate a response body with a maximum file size of 6 MB.
+-   Your project must comply with all applicable [{{ PRODUCT }} {{ PRODUCT_EDGE }} limitations.](/guides/performance/limits)
+
+## NodeJS native extensions {/*nodejs-native-extensions*/}
+
+In a lot of scenarios, NodeJS native extensions might be required to perform specific tasks related to your application.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks/limits.md:16]:
```suggestion
In a lot of scenarios, NodeJS native extensions might be required in order to perform specific tasks related to your application.
```

File: src/pages/guides/sites_frameworks/limits.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+```js
+import { tmpdir } from 'os';
+import * as path from 'path';
+const tmpFilePath = path.join(tmpdir(), 'temp-upload.jpg');
+```
+
+
+Another thing to keep in mind is that "tmp" directory is ephemeral, meaning that it gets reset/recycled. If you store a
+file in "tmp", it most likely won’t be available in the next request. That’s why you’ll need to use external services
+to store permanent file storage. These external services can be Amazon S3, Google Cloud Storage, or any other storage.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks/limits.md:82]:
```suggestion
to store permanent file storage. These external services can be Amazon S3, Google Cloud Storage, or any other storage service.
```

File: src/pages/guides/sites_frameworks/limits.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+```
+2022-08-08T13:47:13Z - internal error - Error in xdn-deploy-lambda: Your production build exceeds the maximum allowed size of 50 MB (compressed) / 250 MB (uncompressed).
+The current size is 51.19 MB (compressed).
+Please ensure that list of dependencies in package.json contains only those packages that are needed at runtime.
+Move all build-time dependencies such as webpack, babel, etc... to devDependencies, rerun npm | yarn install, and try to deploy again.
+```
+
+Following are the possible fixes that would help you reduce serverless bundle size by better engineering. If none of these does it, feel free to raise an issue on [{{ PRODUCT }} Forums](https://forum.layer0.co).
+
+### Possible Fix [1]: Segregating devDependencies from dependencies {/*possible-fix-1-segregating-devdependencies-from-dependencies*/}
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks/limits.md:97]:
```suggestion
### [1]: Segregating devDependencies from dependencies {/*possible-fix-1-segregating-devdependencies-from-dependencies*/}
```

File: src/pages/guides/sites_frameworks/limits.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+-   "babel": "7.12.7",
+-   "jest": "28.1.3"
++ },
++ "devDependencies": {
++   "babel": "7.12.7",
++   "jest": "28.1.3"
+}
+```
+
+### Possible Fix [2]: Segregating assets from serverless bundle {/*possible-fix-2-segregating-assets-from-serverless-bundle*/}
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks/limits.md:114]:
```suggestion
### [2]: Segregating assets from serverless bundle {/*possible-fix-2-segregating-assets-from-serverless-bundle*/}
```

File: src/pages/guides/sites_frameworks/limits.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+Now, you can update your code references from importing the assets to referencing the static path, such as:
+
+```diff
+- import myImage from 'public/images/Image1.png'
+...
+- <div><img src={myImage}/></div>
++ <div><img src="/assets/images/Image1.png"/></div>
+```
+
+### Possible Fix [3]: Computing which node_modules be included in the serverless bundle {/*possible-fix-3-computing-which-node_modules-be-included-in-the-serverless-bundle*/}
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks/limits.md:135]:
```suggestion
### [3]: Computing which node_modules be included in the serverless bundle {/*possible-fix-3-computing-which-node_modules-be-included-in-the-serverless-bundle*/}
```

**Review by RaeesBhatti — CHANGES_REQUESTED** [review]
Thank you putting this together @jsernaque . I left some comments for you consideration

File: src/pages/guides/sites_frameworks.md [hunk 11]
```diff
@@ -2,213 +2,42 @@
 title: Sites (Frameworks)
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:2]:
```suggestion
title: Sites
```

File: src/pages/guides/sites_frameworks.md [hunk 12]
```diff
@@ -2,213 +2,42 @@
...
 ---
 
-{{ PRODUCT }} {{ PRODUCT_PLATFORM }} allows you to quickly integrate your headless Jamstack applications through one of our framework integrations. This integration allows {{ PRODUCT }} to further improve your website's performance through the server-side rendering (SSR) of your website's JavaScript. 
+{{ PRODUCT }} {{ PRODUCT_PLATFORM }}, which provides optimal performance for your Jamstack sites or applications, is powered by [{{ PRODUCT }} {{ PRODUCT_EDGE }}](/guides/performance). This means that you will be able to develop your application at a faster rate and it empowers you to define performance optimizations within your application code.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:5]:
```suggestion
{{ PRODUCT }} {{ PRODUCT_PLATFORM }} is a Jamstack platform built from the group up to run high-performance websites and web apps. It is the only Jamstack platform with an integrated world-class CDN [{{ PRODUCT }} {{ PRODUCT_EDGE }}](/guides/performance) and state-of-the-art [web security suite](/guides/security). {{ PRODUCT }} {{ PRODUCT_PLATFORM }} enables you to build your web apps faster and leverage a global CDN in the application code to achieve the best performance in production. 
```

File: src/pages/guides/sites_frameworks.md [hunk 13]
```diff
@@ -2,213 +2,42 @@
...
+{{ PRODUCT }} {{ PRODUCT_PLATFORM }}, which provides optimal performance for your Jamstack sites or applications, is powered by [{{ PRODUCT }} {{ PRODUCT_EDGE }}](/guides/performance). This means that you will be able to develop your application at a faster rate and it empowers you to define performance optimizations within your application code. 
 
-## Availablity {/*availablity*/}
+Build upon these performance improvements through Server-Side Rendering (SSR). SSR allows you to compute code embedded within your web application using our Serverless Compute workers. Processing your code  within the {{ PRODUCT }} cloud eliminates the need for web servers to run that code. More importantly, it improves performance  through caching without compromising the dynamic nature of these requests. Responses served from cache will achieve a comparable Time to First Byte (TTFB) as static cached content.
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:7]:
```suggestion
Build upon these performance improvements through Server-Side Rendering (SSR). {{ PRODUCT }} {{ PRODUCT_PLATFORM }} takes an SSR-first approach and enables you to render your web apps using our Serverless Compute at any scale. This means that you don't have to maintain and scale the backend server infrastructure and your teams can focus on shipping features your customers will love. {{ PRODUCT }} {{ PRODUCT_PLATFORM }} improves real-world performance through [fine-grained caching](/guides/performance/caching#cache-key), allowing you to cache highly dynamic content. Responses served from the CDN cache will achieve a comparable Time to First Byte (TTFB) as static cached content.
```

File: src/pages/guides/sites_frameworks.md [hunk 14]
```diff
@@ -2,213 +2,42 @@
...
-In a lot of scenarios, NodeJS native extensions might be required to perform specific tasks related to your application.
-For example, you might need to use [OpenCV](https://github.com/peterbraden/node-opencv) to perform some checks on an image before making it publicly available.
-Or you might need to use extensions like [`node-microtime`](https://github.com/wadey/node-microtime) for finer-grained performance analysis.
+## Compute Regions {/*regions*/}
 
-When {{ PRODUCT_NAME }} bundles your application for deployment, we also do some "tree-shaking" to remove unnecessary files in your build.
-This makes the bundle size smaller and more efficient to load on our serverless platform during a cold-start.
-But it could have unintended consequences where we might strip away native extension binaries required for your
-application to function.
+{{ PRODUCT }} allows you to cache and deliver your content through any of our points of presence (POPs) around the world. However, your serverless code will be computed within one of the following global regions:
```
Comment (RaeesBhatti) [inline src/pages/guides/sites_frameworks.md:25]:
```suggestion
{{ PRODUCT }} allows you to cache and deliver your content through any of our points of presence (POPs) around the world. However, your serverless code will run in one of the following global regions:
```

**Review by RaeesBhatti — DISMISSED** [review]
Looks excellent! Suggested some changes that I think will resonate more with web devs.

**Review by RaeesBhatti — APPROVED** [review]
LGTM! Thank you!
